### PR TITLE
feat(auth): [2/4] add global --user flag and tw account list/use commands

### PR DESCRIPTION
**Pull request by scottlovegrove**
**Part 2 of 4 stacked PRs.** Builds on #217.

When all 4 land, the integration branch `scottl/multi-user-base` merges into `main` as one release.

## Order

- #217 — refactor: foundation
- **#218 this PR** — feat: `--user` flag and `tw account list/use`
- #219 — fix: per-account `current_workspace`
- #220 — feat: postinstall v1 → v2 migration

## Summary

Exposes the user-facing surface on top of part 1's storage layer.

### Global `--user <id|email>` (`lib/global-args.ts`, `src/index.ts`)

Selects which stored account this invocation runs against. `undefined` falls back to default / single-stored.

- `parseTwLocalFlags` extends with `user`. The flag honours `--` (tokens after the terminator are positional).
- `stripUserFlag` removes `--user <ref>` from argv before Commander sees it, since Commander has no global-option attachment.
- `validateUserFlag` runs pre-Commander to reject:
  - bare `--user` (no value)
  - `--user=` (empty value)
  - `--user <known-subcommand>` (typo: forgot the value)
  Index.ts surfaces the error as `USER_FLAG_INVALID` CliError.

### `tw account list / use` (`commands/account/`)

- `account list` shows all stored accounts with a default marker. `--json` emits the full record. Named `account` because `tw user` already shows current Twist API user info.
- `account use <ref>` calls `setDefaultUserId` and confirms.

### `tw auth status` (`commands/auth/status.ts`)

- Reads config + metadata in parallel; renders an env-source marker for `TWIST_API_TOKEN` and a default marker for the active user.
- Lists other stored accounts when more than one is present.
- `--json` envelope grows `source`, `isDefault`, and `storedAccounts[]`.

### `tw auth logout` (`commands/auth/logout.ts`)

- Now respects `--user <ref>` so multi-user installs can scope the logout. Passes the ref through to `clearApiToken`.

### SKILL_CONTENT

Documents `--user` + `tw account` subcommands. `skills/twist-cli/SKILL.md` regenerated via `npm run sync:skill`.

### Tests

- New `global-args.user.test.ts` covers `parseGlobalArgs` / `stripUserFlag` / `validateUserFlag` for the bare / empty / subcommand-lookalike / terminator cases.

## Test plan

- [x] `npm run type-check`
- [x] `npm run lint:check`
- [x] `npm test` — 556 pass
- [x] `tw --user inbox` → `USER_FLAG_INVALID` with hint
- [x] `tw --user` → `USER_FLAG_INVALID` with hint
- [x] `tw account --help` lists `list` and `use`

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -199,21 +203,45 @@ if (process.argv[2] === 'completion-server') {
...
         process.exitCode = 1
-    })
-    .finally(() => {
-        stopEarlySpinner()
-    })
+    } else {
+        // Strip `--user <ref>` from argv before Commander sees it. The value
+        // is still available via `getRequestedUserRef()` (read from the
+        // cached parse of the *original* argv).
+        const cleaned = stripUserFlag(userArgs)
```
Comment (doistbot) [inline src/index.ts:224]: [P1] The global args store is lazily initialized. Since `process.argv` is mutated here before any calls (like `isJsonMode()` or `getRequestedUserRef()`) trigger caching, the store will eventually initialize using the stripped `process.argv` when Commander hooks run. This causes a valid `--user <ref>` to silently fall back to `undefined` for the command execution. Call `getRequestedUserRef()` right before this line to force the cache to populate with the original arguments before stripping.

File: src/index.ts [hunk 2]
```diff
@@ -199,21 +203,45 @@ if (process.argv[2] === 'completion-server') {
...
+// `--user --json`, `--user=`, and `--user inbox` (forgot the value) produce a
+// clear CliError with hints instead of Commander's terser usage message.
+{
+    const knownCommands = new Set<string>([
+        ...Object.keys(commands),
+        ...Object.keys(commandAliases),
+    ])
+    const userArgs = process.argv.slice(2)
+    const validation = validateUserFlag(userArgs, knownCommands)
+    if (!validation.ok) {
```
Comment (doistbot) [inline src/index.ts:216]: [P2] This validation branch skips `parseAsync()`, but the command loader above may already have called `startEarlySpinner()`. For inputs like `tw --user inbox`, the loader starts the spinner, validation fails here, and nothing stops it. Move the `--user` validation/stripping ahead of command loading, or call `stopEarlySpinner()` before returning this error.

File: src/index.ts [hunk 3]
```diff
@@ -199,21 +203,45 @@ if (process.argv[2] === 'completion-server') {
...
+        const err = new CliError('USER_FLAG_INVALID', validation.message, validation.hints, 'info')
+        console.error(isJsonMode() ? formatErrorJson(err) : formatError(err))
         process.exitCode = 1
-    })
-    .finally(() => {
-        stopEarlySpinner()
-    })
+    } else {
+        // Strip `--user <ref>` from argv before Commander sees it. The value
```
Comment (doistbot) [inline src/index.ts:221]: [P1] Stripping the flag here makes Commander happy, but the selected account still is not used by normal command execution. `getRequestedUserRef()` is only consumed by `auth/logout`; all API-backed paths still go through `getApiToken()`/`resolveActiveUser()` with no ref, so `tw --user other@example.com inbox` will still run as the default account. Please plumb the requested ref into auth resolution centrally before advertising `--user` as a global selector.

File: src/index.ts [hunk 4]
```diff
@@ -199,21 +203,45 @@ if (process.argv[2] === 'completion-server') {
...
+{
+    const knownCommands = new Set<string>([
+        ...Object.keys(commands),
+        ...Object.keys(commandAliases),
+    ])
+    const userArgs = process.argv.slice(2)
+    const validation = validateUserFlag(userArgs, knownCommands)
+    if (!validation.ok) {
+        const err = new CliError('USER_FLAG_INVALID', validation.message, validation.hints, 'info')
+        console.error(isJsonMode() ? formatErrorJson(err) : formatError(err))
```
Comment (doistbot) [inline src/index.ts:218]: [P2] The manual error logging and exit logic here duplicates the global handler on line 234 and bypasses `stopEarlySpinner()`. Instead of logging and setting `process.exitCode` manually, consider wrapping the validation block and `program.parseAsync()` in a unified `try/catch` block. This allows you to simply `throw new CliError(...)` here and share the formatting, exit code, and spinner cleanup logic.

File: src/lib/global-args.ts [hunk 5]
```diff
@@ -181,3 +203,89 @@ export const shouldDisableSpinner = createSpinnerGate({
...
+export function validateUserFlag(
+    argv: string[],
+    knownCommands: ReadonlySet<string>,
+): UserFlagValidation {
+    const terminatorIndex = argv.indexOf('--')
+    const scanRange = terminatorIndex === -1 ? argv : argv.slice(0, terminatorIndex)
+    const sawUserFlag = scanRange.some((a) => a === '--user' || a.startsWith('--user='))
+    if (!sawUserFlag) return { ok: true, ref: undefined }
+
+    const ref = parseGlobalArgs(argv).user
```
Comment (doistbot) [inline src/lib/global-args.ts:275]: [P3] `parseGlobalArgs` delegates to `parseCoreGlobalArgs` under the hood, which unnecessarily parses all of the `@doist/cli-core` flags just to throw them away. Since `validateUserFlag` and `parseTwLocalFlags` are in the same file, you can call `parseTwLocalFlags(argv).user` directly to avoid the redundant parsing overhead.

File: src/commands/auth/status.ts [hunk 6]
```diff
@@ -19,21 +21,68 @@ export async function showStatus(options: { json?: boolean }): Promise<void> {
         throw error
     }
 
+    // One config read covers both the stored-users list and the
+    // default-id lookup; metadata fetch runs in parallel so cold status
+    // doesn't pay for sequential FS hops.
+    const [metadata, config] = await Promise.all([getAuthMetadata(), getConfig()])
```
Comment (doistbot) [inline src/commands/auth/status.ts:27]: [P2] `getAuthMetadata()` already resolves the active user via `resolveActiveUser()`, which reads the config for stored-account auth. Adding a separate `getConfig()` here makes `tw auth status` hit the config file one extra time on every successful run. Consider a combined helper that returns both auth metadata and the stored-user/default snapshot from a single config read.

File: src/commands/account/index.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import { Command } from 'commander'
+import { listAccountsCommand } from './list.js'
+import { useAccountCommand } from './use.js'
+
+export function registerAccountCommand(program: Command): void {
```
Comment (doistbot) [inline src/commands/account/index.ts:5]: [P2] The new `tw account list` and `tw account use` commands are the core functional additions in this diff, but they are currently untested. Consistent with the repository convention of colocating tests (as documented in AGENTS.md, e.g., `src/commands/config/config.test.ts`), consider adding `src/commands/account/account.test.ts` to cover the listing behavior, default account markers, and missing-ref error paths.

File: src/lib/global-args.user.test.ts [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    })
+
+    it('returns ok: true with the value', () => {
+        expect(validateUserFlag(['--user', 'me@example.com', 'inbox'], KNOWN)).toEqual({
+            ok: true,
+            ref: 'me@example.com',
+        })
+    })
+
+    it('rejects bare --user', () => {
```
Comment (doistbot) [inline src/lib/global-args.user.test.ts:66]: [P2] These invalid `--user` cases only assert `ok === false`, so they won't catch regressions in the new user-facing guidance this PR is adding (`message`/`hints`, and the `USER_FLAG_INVALID` path in `index.ts`). Please either assert the returned `message`/`hints` here or add an entrypoint-level test that verifies the formatted error for bare/empty `--user`.

File: src/commands/auth/status.ts [hunk 9]
```diff
@@ -19,21 +21,68 @@ export async function showStatus(options: { json?: boolean }): Promise<void> {
...
+    const storedUsers = getStoredUsers(config)
+    const defaultUserId = getDefaultUserId(config)
+    const userIdStr = String(user.id)
+
     if (options.json) {
-        console.log(formatJson({ id: user.id, email: user.email, name: user.name }))
+        console.log(
```
Comment (doistbot) [inline src/commands/auth/status.ts:33]: [P2] `auth status` now has several new branches in both its JSON contract and text rendering (`source`, `isDefault`, `storedAccounts`, and env-vs-default marker precedence), but this PR adds no coverage for them. Please add focused tests that stub metadata/config and pin both the `--json` payload and the text output for default-account vs `TWIST_API_TOKEN` scenarios.

File: src/commands/account/index.ts [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+        .description('Manage stored Twist accounts (multi-user)')
+
+    account
+        .command('list')
+        .description('List all stored Twist accounts')
+        .option('--json', 'Output as JSON')
+        .action(listAccountsCommand)
+
+    account
+        .command('use <ref>')
```
Comment (doistbot) [inline src/commands/account/index.ts:17]: [P2] `account use` mutates local CLI state, but this new mutating command doesn't provide the repo-standard `--json` output path for scripting. Please add `--json` here (ideally by wiring a `MutationOptions`-style options object through to the handler) and emit a minimal status object instead of text-only output.

File: src/lib/skills/content.ts [hunk 11]
```diff
@@ -37,7 +41,9 @@ tw update                        # Update CLI to latest version
 tw changelog                     # Show recent changelog entries
 \`\`\`
 
-Stored auth uses the system credential manager when available. If secure storage is unavailable, \`tw\` warns and falls back to \`~/.config/twist-cli/config.json\`. \`TWIST_API_TOKEN\` always takes priority over the stored token, and legacy plaintext config tokens are migrated automatically when secure storage is available.
+Multiple Twist accounts can be authenticated simultaneously. Each account's token is stored in a per-account keyring slot (\`user-<id>\`), and per-account state (\`current_workspace\`) is tracked separately so \`tw --user other@example.com inbox\` doesn't reuse the previous account's workspace selection. Without \`--user\`, the default account is used; if only one account is stored, it's implicit.
```
Comment (doistbot) [inline src/lib/skills/content.ts:44]: [P2] This skill text is ahead of the implementation. The runtime still reads/writes the top-level `currentWorkspace`, and the automatic postinstall migration is called out as a later stacked PR. AGENTS requires `SKILL_CONTENT` to stay accurate to current CLI behavior, so please reword or defer these claims until those follow-up changes land.

**Review by doistbot — COMMENTED** [review]
This PR introduces the user-facing `--user` global flag and `tw account` commands, successfully building the CLI interface on top of the new multi-user storage foundation. The changes are well-structured and thoughtfully extend existing commands like `auth status` and `logout` to accommodate multiple accounts. A few structural adjustments are needed to ensure the `--user` flag is properly plumbed into the global auth resolution and caching, alongside some consolidation of error handling, additions to test coverage for the new commands, and aligning the skill documentation and `--json` support with current CLI conventions.

<sub>[Share Feedback](https://github.com/Doist/doistbot-app/issues/new?template=review-feedback.yml&title=Review+feedback+for+Doist%2Ftwist-cli&pr_url=https%3A%2F%2Fgithub.com%2FDoist%2Ftwist-cli%2Fpull%2F218&datadog_url=https%3A%2F%2Fapp.datadoghq.com%2Flogs%3Fquery%3Dservice%253A%2528doistbot-app-sandbox%2520OR%2520doistbot-app%2529%2520%2540meta.deliveryId%253Ac36c4920-4d67-11f1-8afc-d2c5f8b1e6b2%26agg_m%3Dcount%26agg_m_source%3Dbase%26agg_t%3Dcount%26clustering_pattern_field_path%3Dmessage%26cols%3Dhost%252Cservice%26fromUser%3Dtrue%26messageDisplay%3Dinline%26refresh_mode%3Dsliding%26storage%3Dhot%26stream_sort%3Ddesc%26viz%3Dstream%26live%3Dtrue) • [Review Logs](https://app.datadoghq.com/logs?query=service%3A%28doistbot-app-sandbox%20OR%20doistbot-app%29%20%40meta.deliveryId%3Ac36c4920-4d67-11f1-8afc-d2c5f8b1e6b2&agg_m=count&agg_m_source=base&agg_t=count&clustering_pattern_field_path=message&cols=host%2Cservice&fromUser=true&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=stream&live=true)</sub>

**Comment by scottlovegrove** [comment]
Closing — multi-user auth will come via cli-core eventually.
